### DISCUSSION TITLE
Specify Public IP for UDPMux

### DISCUSSION THREAD

**Discussion by Wicpar** [webrtc-rs/webrtc#370] [Q&A] [answered]
Hi,

I am using a muxed udp connection however there doesn't seem to be a way to indicate the public ip address.
The `Conn` trait has a `remote_addr` method, is it the one that needs to be implemented to specify the external IP ?

Currently all accesses to the servers use turn from the server side when they could use the host type directly if they had the IP, and it generates quite high data costs.

**Comment by k0nserv** [comment, marked as answer]
This is what we do in combination with muxxed UDP where `nat_ips` is `Vec<String>`.

```rust
settings.set_nat_1to1_ips(nat_ips, webrtc::RTCIceCandidateType::Host);
```

 We only do IPv4 atm. I think there's a bit more work to support both an IPv4 and IPv6 address. But if you have dual stack server feel free to try specifying both

**Comment by Wicpar** [comment]
Thanks a lot ! I found it after some time and didn't get around to closing the issue, is there a guide section somewhere where we could document these kind of things ?

**Comment by k0nserv** [comment]
It could definitely be documented better. The examples are probably the closest thing to this atm. Long term it'd be cool to have the webrtc-rs book but then someone needs to write that 😅
